src/algebra/homology/null_homotopic.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2021 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import algebra.homology.homotopy
+
+/-!
+# Null homotopic chain complex maps
+
+-/
+
+open category_theory
+open category_theory.category
+
+namespace chain_complex
+
+variables {V : Type*} [category V] [preadditive V]
+
+/-- Given a sequence h of morphisms `C_n ⟶ D_{n+1}`, this is the chain
+complex morphism C ⟶ D given by the formula `dh+hd`. -/
+@[simps]
+def null_homotopic_map {C D : chain_complex V ℕ}
```

Comment:
Can this definition be stated in a way that generalizes `chain` and `cochain` complexes? We've tried to do a bunch of stuff for `homological_complex`. But sometimes it's painful to generalize.
At the same time, we don't want a lot of copies of this definition (for (co)chain complexes), for complexes indexed by ℕ, by ℤ, ...

Reply:
Ok, I will try see to what extent I can make it work, certainly with something more general than ℕ, and maybe also homological_complex... Thanks for the suggestion.

Reply:
Note that your current definition is doing something special for `n = 0`. So generalizing away from `ℕ` might already be a bit tricky. But somehow, the `complex_shape` stuff was designed to deal with such issues. Hopefully we made the right choices, so that it also works here.

Reply:
Indeed, it works, see next commit. I will test these definitions a little bit more, and add documentation later. (Presumably, everything should be moved to algebra/homology/homotopy.lean instead of a separate file.)

Reply:
Great! This is looking good. And yes, I think this can move to `homotopy.lean`.